cmds.go

```diff
@@ -198,21 +198,14 @@ func checkEnvironment(socket, certPath string) bool {
 }
 
 func printExport(socket, certPath string) {
-	for name, value := range exports(socket, certPath) {
-		switch filepath.Base(os.Getenv("SHELL")) {
-		case "fish":
-			if value == "" {
-				fmt.Printf("    set -e %s\n", name)
-			} else {
-				fmt.Printf("    set -x %s %s\n", name, value)
-			}
-		default: // default command to export variables POSIX shells, like bash, zsh, etc.
-			if value == "" {
-				fmt.Printf("    unset %s\n", name)
-			} else {
-				fmt.Printf("    export %s=%s\n", name, value)
-			}
-		}
+	env := exports(socket, certPath)
+	switch filepath.Base(os.Getenv("SHELL")) {
+	case "fish":
+		fmt.Printf("set -x DOCKER_TLS_VERIFY %s\nset -x DOCKER_CERT_PATH %s\nset -x DOCKER_HOST %s\n",
+			env["DOCKER_TLS_VERIFY"], strings.Replace(env["DOCKER_CERT_PATH"], ` `, `\ `, -1), env["DOCKER_HOST"])
+	default:
+		fmt.Printf("export DOCKER_TLS_VERIFY=%s DOCKER_CERT_PATH=%s DOCKER_HOST=%s\n",
```

Comment:
Pretty strongly -1 on a hard-coded list like this, especially when we already have key-value pairs telling us the full list of variables we're expected to set.  This would mean that anyone adding a new variable to the list of vars we set would have to know all the places they need to update, which is error-prone, especially if we look to add more shells here.